packages/components/src/number-control/index.js

```diff
@@ -43,10 +43,13 @@ export function NumberControl(
 	const baseStep = isStepAny ? 1 : parseFloat( step );
 	const baseValue = roundClamp( 0, min, max, baseStep );
 	const constrainValue = ( value, stepOverride ) => {
+		let val = parseFloat( value );
```

Comment:
Overall I think this is a nice option! Two main pieces of feedback:

1. If folks agree to add this, let's make sure functionality works similarly in other controls. Potentially in follow ups. I think we may want to not allow notation like `2e4` when units aren't allowed. 
2. Have we considered only accepting postfix strings of "px", "em", "rem", "vw", "vh"? I'd recommend ignoring the value if an unknown postfix is given.

<img width="156" alt="CleanShot 2022-02-01 at 15 00 44@2x" src="https://user-images.githubusercontent.com/1270189/152066009-0f81999c-37db-4d64-bcc2-fd1a6b667ba2.png">

The reason I'm asking, is that `parseInt` and `parseFloat` can sometimes return surprising values. If the value isn't a string, toString() is called, and JS tries its best to parse as much of the string as it can and return a value instead of `NaN` like other languages. A classic weird example is a value like `parseInt( 0.0000007 )` returning 7. 

<img width="244" alt="parseInt oddity" src="https://user-images.githubusercontent.com/1270189/152066524-abcedd9a-3a68-4f61-b298-dd0a0c6d036e.png">

Reply:
Thank you for taking a look!

> If folks agree to add this, let's make sure functionality works similarly in other controls. Potentially in follow ups. I think we may want to not allow notation like 2e4 when units aren't allowed.

Agreed on both counts ➕ 

> Have we considered only accepting postfix strings of "px", "em", "rem", "vw", "vh"? I'd recommend ignoring the value if an unknown postfix is given.

Hm -- I think you're right, and this is probably the best approach for preventing weird edge cases. But now I'm not so sure it's reasonable to expect the `NumberControl` to know about units outside of the context of the `UnitControl`. There may not be a clean way to do this as I had hoped 🤔